[IMP] barcode: split docs + hone in on language (#5592)
odoo/documentation: content/applications/inventory_and_mrp/inventory/barcode/operations/gs1_usage.rst

=================
GS1 barcode usage
=================

.. _barcode/operations/gs1_usage:

.. |AI| replace:: :abbr:`A.I. (Application Identifier)`
.. |GTIN| replace:: :abbr:`GTIN (Global Trade Item Number)`

Setting up GS1 barcodes in Odoo enables businesses to configure their products and make them

> globally recognized across the supply chain. In Odoo, GS1 barcodes streamline various warehouse

Author: I rewrote the intro! What do you think?

> GS1 barcodes are a standardized language for barcode scanners helping businesses with international
supply chain operations. Use Odoo *Barcode* to interpret and print GS1 barcodes, making warehouse
operations, like receiving and picking, more efficient.

processes, such as receiving packages, by automating data entry and reducing processing times.

The following sections contain examples of how Odoo uses GS1 barcodes provided by the business to
identify common warehouse items and automate certain warehouse workflows.

.. important::

>    Odoo **does not** create GS1 barcodes. Businesses must provide a unique Global Trade Item Number

Reviewer: ```suggestion
   Odoo **does not** create GS1 barcodes. Businesses must purchase a unique Global Trade Item Number
```

   (GTIN) from GS1. Then, they can combine their existing GS1 barcodes with product and supply chain
   information (also provided by GS1) to create barcodes in Odoo.

.. seealso::
   - `Purchase GTINs <https://www.gs1.org/standards/get-barcodes>`_
   - :ref:`GS1 nomenclature <barcode/operations/gs1>`

.. _barcode/operations/gs1-lots:

> Product + quantity + lot

Author: ```suggestion
Configure barcodes for product, quantity, and lots
```

   Scanner` section. Enter the 14-digit GTIN into the :guilabel:`Barcode` column, then click
   :guilabel:`Save`.

   .. image:: gs1_usage/product-barcodes-page.png
      :align: center
      :alt: View the Product Barcodes page from inventory settings.

.. _barcode/operations/lot-setup-on-product:

After activating lots and serial number tracking in the settings page, specify that this feature is
to be applied on each product by navigating to the :guilabel:`Inventory` tab on the product form.
Under :guilabel:`Tracking`, choose the :guilabel:`By Lots` radio button.

.. image:: gs1_usage/track-by-lots.png
   :align: center
   :alt: Enable product tracking by lots in the "Inventory" tab of the product form.

Scan barcode on receipt
-----------------------

To ensure accurate lot interpretation in Odoo on product barcodes scanned during a receipt
operation, navigate to the :menuselection:`Barcode` app to manage the :ref:`receipt picking process
<barcode/operations/scan-received-products>`.

From the :guilabel:`Barcode Scanning` dashboard, click the :guilabel:`Operations` button, then the

> :guilabel:`Receipts` button to view the list of vendor receptions to process. Receipts generated

Reviewer: ```suggestion
:guilabel:`Receipts` button to view the list of vendor receipts to process. Receipts generated
```
Which sounds better?

   scanner, below is an example barcode for the fifty Fuji apples in Lot 2.

   .. list-table::
      :widths: 50 50
      :header-rows: 1
      :stub-columns: 1

      * - 50 Fuji apples in Lot0002
        -
      * - 2D Matrix
        - .. image:: gs1_usage/fuji-apples-barcode.png
             :alt: 2D matrix of GS1 barcode of 50 fuji apples with an assigned lot number.
      * - |AI| (product)
        - 01
      * - GS1 Barcode (product)
        - 20611628936004
      * - |AI| (quantity)
        - 30
      * - GS1 Barcode (quantity)
        - 00000050
      * - |AI| (lot)
        - 10
      * - GS1 Barcode (lot #)
        - LOT0002
      * - Full GS1 barcode
        - 01206116289360043 000000050 10LOT0002

   :ref:`If the configuration is correct <barcode/operations/troubleshooting>`, `50/50`
   :guilabel:`Units` processed will be displayed and the :guilabel:`Validate` button turns green.
   Click the :guilabel:`Validate` button to complete the reception.

   .. image:: gs1_usage/receive-50-apples.png
      :align: center
      :alt: Scan the barcode for a product on the reception picking page in the *Barcode* app.

.. _barcode/operations/quantity-ex:

> Product + non-unit quantity

Reviewer: Same note for the similar heading above. But if you want to stick with this style heading then feel free, it's up to you

> Product + non-unit quantity

Author: ```suggestion
Configure barcode for product and non-unit quantity
```

=================
GS1 barcode usage
=================

.. _barcode/operations/gs1_usage:

.. |AI| replace:: :abbr:`A.I. (Application Identifier)`
.. |GTIN| replace:: :abbr:`GTIN (Global Trade Item Number)`

> GS1 barcodes are a standardized language for barcode scanners helping businesses with international

Reviewer 2: - singular/plural mismatch
- *language* is the wrong word here I think?
- this first sentence is a bit clunky 🥴

Author: what do u think about my intro rewrite?
> 
GS1 barcodes provide a standardized format that barcode scanners can interpret. They encode
information in a :ref:`specific structure recognized globally <barcode/operations/gs1>`, allowing
scanners to understand and process supply chain data consistently.

> Odoo *Barcode* interprets and prints GS1 barcodes, automating product identification and tracking
in warehouse operations such as receiving, picking, and shipping.

Reviewer 2: good rewrite
one revision: 
`pecific structure recognized globally` --> `a specific, globally recognized structure`

supply chain operations. Use Odoo *Barcode* to interpret and print GS1 barcodes, making warehouse
operations, like receiving and picking, more efficient.

The following sections contain examples of how Odoo uses GS1 barcodes provided by the business to
identify common warehouse items and automate certain warehouse workflows.

.. important::
   Odoo **does not** create GS1 barcodes. Businesses must purchase a unique Global Trade Item Number
   (GTIN) from GS1. Then, they can combine their existing GS1 barcodes with product and supply chain
   information (also provided by GS1) to create barcodes in Odoo.

.. seealso::
   - `Purchase GTINs <https://www.gs1.org/standards/get-barcodes>`_
   - :ref:`GS1 nomenclature <barcode/operations/gs1>`

.. _barcode/operations/gs1-lots:

Configure barcodes for product, quantity, and lots
==================================================

> To build a GS1 barcode for a box that contains a product, number of units in it, and the lot number,

Reviewer 2: The way this is phrased implies that all three variables are necessary in a GS1 barcode. I don't think they are?

What about information like batch number, expiration dates, etc., which can also be included in GS1's?

I'd rephrase this intro sentence so it's not forcing a specific scenario:

```suggestion
To build a GS1 barcode for a shipment, use the following barcode patterns and Application Identifiers (A.I.) in the table below:
```

For a future PR, especially since the apple is a good example: I wonder if you can include expiration dates too here?

| Product    | Global Trade Item Number | 01   | (01)(\\d{14})                    | :guilabel:`Barcode` field on product form|
|            | (GTIN)                   |      |                                  |                                          |
+------------+--------------------------+------+----------------------------------+------------------------------------------+
| Quantity   | Variable count of items  | 30   | (30)(\\d{0,8})                   | :guilabel:`Units` field on transfer form |
+------------+--------------------------+------+----------------------------------+------------------------------------------+
| Lot Number | Batch or lot number      | 10   | (10)([!"%-/0-9:-?A-Z_a-z]{0,20}) | :guilabel:`Lot` on Detailed Operations   |
|            |                          |      |                                  | pop-up                                   |
+------------+--------------------------+------+----------------------------------+------------------------------------------+

.. _barcode/operations/lot-setup:

Configuration
-------------

> To track products using lots, first enable the :ref:`Lots and Serial Numbers
> <inventory/management/track_products_by_lots>` feature. To do so, navigate to
> :menuselection:`Inventory app --> Configuration --> Settings`. Next, under the
> :guilabel:`Traceability` heading, check the box for :guilabel:`Lots & Serial Numbers`.

Reviewer 2: Condense
```suggestion
First, :ref:`enable product tracking using lots <inventory/management/track_products_by_lots>` by
navigating to :menuselection:`Inventory app --> Configuration --> Settings`, and checking the box
for :guilabel:`Lots & Serial Numbers` under the :guilabel:`Traceability` heading.
```

.. example::
   On the receipt in the *Barcode* app, receive an order for `52.1 kg` of peaches by scanning the
   barcode containing the |GTIN| and quantity of peaches in kilograms.

   .. list-table::
      :widths: 50 50
      :header-rows: 1
      :stub-columns: 1

      * - 52.1 kg of Peaches
        -
      * - 2D Matrix
        - .. image:: gs1_usage/peaches-barcode.png
             :alt: 2D matrix of GS1 barcode of 52.1 kg of peaches.
      * - |AI| (product)
        - 01
      * - GS1 Barcode (product)
        - 00614141000012
      * - |AI| (kg, 1 decimal point)
        - 3101
      * - GS1 Barcode (quantity)
        - 000521
      * - Full GS1 barcode
        - 0100614141000012 3101000521

   :ref:`If the configuration is correct <barcode/operations/troubleshooting>`, `52.1 / 52.1`
   :guilabel:`kg` will be displayed and the :guilabel:`Validate` button turns green. Finally, press
   :guilabel:`Validate` to complete the validation.

   .. image:: gs1_usage/scan-barcode-peaches.png
      :align: center
      :alt: Scan barcode screen for a reception operation in the Barcode app.

Verify product moves
====================

> For additional verification purposes, the quantities of received products are also recorded on the

Reviewer 2: ```suggestion
For additional verification, the quantities of received products are also recorded on the
```

=================
GS1 barcode usage
=================

.. _barcode/operations/gs1_usage:

.. |AI| replace:: :abbr:`A.I. (Application Identifier)`
.. |GTIN| replace:: :abbr:`GTIN (Global Trade Item Number)`

Setting up GS1 barcodes in Odoo enables businesses to configure their products and make them
globally recognized across the supply chain. In Odoo, GS1 barcodes streamline various warehouse
processes, such as receiving packages, by automating data entry and reducing processing times.

The following sections contain examples of how Odoo uses GS1 barcodes provided by the business to
identify common warehouse items and automate certain warehouse workflows.

.. important::
   Odoo **does not** create GS1 barcodes. Businesses must provide a unique Global Trade Item Number
   (GTIN) from GS1. Then, they can combine their existing GS1 barcodes with product and supply chain

>    information (also provided by GS1) to create barcodes in Odoo.

Author: @StraubCreative Do you think this makes it clear that GTINs are unique and purchased from GS1?

directly through the :menuselection:`Barcode` app using the :guilabel:`Create` button.

On the list of receipts, click on the warehouse operation (`WH/IN`) and scan product barcodes and
lot numbers with a barcode scanner. The scanned product then appears on the list. Use the
:guilabel:`✏️ (pencil)` button to open a window and manually enter quantities for specific lot
numbers.

.. example::
   After placing a :abbr:`PO (Purchase Order)` for fifty apples, navigate to the associated receipt
   in the *Barcode* app.

   Scan the barcode containing the |GTIN|, quantity, and lot number. For testing with a barcode
   scanner, below is an example barcode for the fifty Fuji apples in Lot 2.

   .. list-table::
      :header-rows: 1
      :stub-columns: 1

      * -
        - 2D Matrix
        - |AI|
          (product)
        - GS1 Barcode (product)
        - |AI|
          (quantity)
        - GS1 Barcode (quantity)
        - |AI| (lot)
        - GS1 Barcode (lot #)
        - Full GS1 barcode
      * - 50 Fuji apples in Lot0002
        - .. image:: gs1_usage/fuji-apples-barcode.png
             :align: center

>              :alt: 2D matrix of GS1 barcode of 50 fuji apples with an assigned lot number.

Author: @samueljlieber Do you have any suggestions about fixing the image width/heights or scaling the barcode? Even at full size, the barcode image is a little small and I'm wondering if it'll be an issue when the window needs to be minimized/resized.

Reviewer 3: I agree the QR code and table headers do appear to be scrunched in this layout. I'd suggest flipping the table axis so there are two columns, giving more room to breathe and accommodate different screen sizes. I added RST here: https://github.com/odoo/documentation/pull/5592/files/73cd08835811c5e774d4ac0e58fad9e38ac48cb2#r1305710601

<img width="942" alt="Screenshot 2023-08-25 at 10 06 49 AM" src="https://github.com/odoo/documentation/assets/36018073/1ab2259d-ff69-4401-983a-41f14a8cb33b">

>    .. list-table::
>       :header-rows: 1
>       :stub-columns: 1
> 
>       * -
>         - 2D Matrix
>         - |AI|
>           (product)
>         - GS1 Barcode (product)
>         - |AI|
>           (quantity)
>         - GS1 Barcode (quantity)
>         - |AI| (lot)
>         - GS1 Barcode (lot #)
>         - Full GS1 barcode
>       * - 50 Fuji apples in Lot0002
>         - .. image:: gs1_usage/fuji-apples-barcode.png
>              :align: center
>              :alt: 2D matrix of GS1 barcode of 50 fuji apples with an assigned lot number.
>         - 01
>         - 20611628936004
>         - 30
>         - 00000050
>         - 10
>         - LOT002
>         - 01206116289360043
>           000000050
>           10LOT0002

Reviewer 3: ```suggestion
   .. list-table::
      :widths: 50 50
      :header-rows: 1
      :stub-columns: 1

      * - 50 Fuji apples in Lot0002
        -
      * - 2D Matrix
        - .. image:: gs1_usage/fuji-apples-barcode.png
             :alt: 2D matrix of GS1 barcode of 50 fuji apples with an assigned lot number.
      * - |AI| (product)
        - 01
      * - GS1 Barcode (product)
        - 20611628936004
      * - |AI| (quantity)
        - 30
      * - GS1 Barcode (quantity)
        - 00000050
      * - |AI| (lot)
        - 10
      * - GS1 Barcode (lot #)
        - LOT0002
      * - Full GS1 barcode
        - 01206116289360043 000000050 10LOT0002
```
